src/providers/cognito.js

```diff
@@ -0,0 +1,23 @@
+export default (options) => {
+  const { region, domain } = options;
+  return {
+    id: 'cognito',
+    name: 'Cognito',
+    type: 'oauth',
+    version: '2.0',
+    scope: 'openid profile email',
+    params: { grant_type: 'authorization_code' },
+    accessTokenUrl: `https://${domain}.auth.${region}.amazoncognito.com/oauth2/token`,
+    authorizationUrl: `https://${domain}.auth.${region}.amazoncognito.com/oauth2/authorize?response_type=code`,
+    profileUrl: `https://${domain}.auth.${region}.amazoncognito.com/oauth2/userInfo`,
```

Comment:
Hey there! Really excited about Amazon Cognito support :D

I was wondering if we should do all domain configuration on providers a bit like this (below).

I was thinking it might be easier for users to take the same approach for each provider, and often is easier to copy/paste the complete domain from the admin UI for the provider when setting it up?

```suggestion
  return {
    id: 'cognito',
    name: 'Cognito',
    type: 'oauth',
    version: '2.0',
    scope: 'openid profile email',
    params: { grant_type: 'authorization_code' },
    accessTokenUrl: `https://${options.domain}/oauth2/token`,
    authorizationUrl: `https://${options.domain}/oauth2/authorize?response_type=code`,
    profileUrl: `https://${options.domain}/oauth2/userInfo`,
```

Reply:
Ooh yeah, I can change that if you want. I'm currently fixing some rebase mess from switching from master to main 😅

Reply:
This probably also solves the custom domain issue I guess?

Reply:
Ah lol I feel your pain!

I did that and had to check StackOverflow to undo the mess I made. 

You might not need this, but if it helps, this is what I ended up doing:

1. `git reset --soft HEAD~1 ` to undo the last checkout, but restoring the changes to the working directory.

   You can do multiple commits here instead of just the last one, but I didn't have many commits in the branch and found it easier to restore and stash each commit separately and then replay them, rather than checking out the last 2-3 commits at once, so I could review the changes more easily.
2. Saving the files it spit out in a stash.
3.  Repeating that for each commit I'd done, saving each commit as a new stash, checking how many I had left to go with `git log`.

Then once I'd saved all the commits as stashes , I did `git reset --hard origin/main` and then applied the stashes back (and committed each one) in order.

It's possible to generate a patch file too I think, but I've only done that a couple of times so found this easier.